src/socket.rs

```diff
@@ -288,7 +324,7 @@ pub enum ConnectError {
     #[error("timeout connecting to server")]
     Timeout,
     #[error("websocket error: {0}")]
-    WebSocketError(#[from] tungstenite::Error),
+    WebSocketError(#[from] Arc<tungstenite::Error>),
```

Comment:
This wouldn't need Arc here if using Arc<Result<T. E>> over Result<T: Clone, Arc<E>>

Reply:
This can't be an `Arc` outside the `Result` because the `tungstenite::Error` is shared between this `ConnectError::WebSocketError` and sent at the `Err(Arc<tunstenite::Error)` in the `Socket::socket` broadcast as discussed in https://github.com/liveview-native/phoenix-channels-client/pull/24#discussion_r1231271855.